app/javascript/hooks/useDebounce.jsx

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useState } from 'react';
+
+export default function useDebounce(value, delay = 500) {
```

Comment:
Debouncing the request tend to make requests only when the user completes typing their input to avoid flooding the server and ruining the UX for users with good connections [CHECK HERE](https://dev.to/otamnitram/throttling-and-debouncing-avoiding-unnecessary-api-calls-2god) and [HERE ](https://www.freecodecamp.org/news/javascript-debounce-example/).

Reply:
This is a problem we have across the app - I'd rather we implement this across the app one shot in a separate PR then just to do it randomly here